### PR TITLE
[ws-manager] change workspace state tracing

### LINKED ISSUE #10590: [tracing] fix workspace running trace
### Bug description

```
We should then remove the old (misguided) "try and persist the span until workspace is running" approach. Central to that code is [traceWorkspace(...)](https://github.com/gitpod-io/gitpod/blob/15cce5f092604dd30df24e86458222815bc73aca/components/ws-manager/pkg/manager/monitor.go#L518) and the [traceID annotation](https://github.com/gitpod-io/gitpod/blob/28635821880f37c5f26a8780380afe621456b4c3/components/common-go/kubernetes/kubernetes.go#L41).
```
https://github.com/gitpod-io/gitpod/pull/10264#issuecomment-1152047424

### Steps to reproduce

n\a

### Workspace affected

_No response_

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sagor999**
## Description

As discussed in https://github.com/gitpod-io/gitpod/pull/10264#issuecomment-1152047424

## Related Issue(s)

Fixes #10590 

## Release Notes

```release-note
none
```

### CODE REVIEW

**Comment by sagor999** [comment]
@csweichel @mads-hartmann 
this creates new spans now, instead of attaching to closed one.
Let me know if you agree on the approach. Not sure if we still wanted to somehow link new span to original span or leave it completely detached. :thinking: 
Will leave this PR in draft while awaiting feedback.

File: components/ws-manager/pkg/manager/imagespec.go [hunk 1]
```diff
@@ -30,16 +30,9 @@ func (m *Manager) GetImageSpec(ctx context.Context, req *regapi.GetImageSpecRequ
...
+		span opentracing.Span
 	)
-	if ok {
-		spanCtx := tracing.FromTraceID(traceID)
-		span = opentracing.StartSpan("GetImageSpec", opentracing.FollowsFrom(spanCtx))
-		ctx = opentracing.ContextWithSpan(ctx, span)
-	} else {
-		span, ctx = tracing.FromContext(ctx, "GetImageSpec")
-	}
+	span, ctx = tracing.FromContext(ctx, "GetImageSpec")
```
Comment (csweichel) [inline components/ws-manager/pkg/manager/imagespec.go:35]: nit:
```suggestion
	span, ctx := tracing.FromContext(ctx, "GetImageSpec")
```

then the `var` block can go away

File: components/ws-manager/pkg/manager/monitor.go [hunk 2]
```diff
@@ -512,25 +509,14 @@ func (m *Monitor) writeEventTraceLog(status *api.WorkspaceStatus, wso *workspace
...
-	spanCtx := tracing.FromTraceID(traceID)
-	if spanCtx == nil {
-		// no trace information available
-		return opentracing.NoopTracer{}.StartSpan("noop")
-	}
-
-	span := opentracing.StartSpan(fmt.Sprintf("/workspace/%s", occasion), opentracing.FollowsFrom(spanCtx))
+// traceWorkspaceState creates a new span that records the phase of workspace
+func (m *Monitor) traceWorkspaceState(state string, wso *workspaceObjects) opentracing.Span {
+	span := opentracing.StartSpan(fmt.Sprintf("/workspace/%s", state))
```
Comment (csweichel) [inline components/ws-manager/pkg/manager/monitor.go:514]: This span is no longer connected to the workspace. We should add the instance/workspace ID as label so that we can correlate.

**Review by csweichel — APPROVED** [review]
Awesome! One step closer to better traces :) 

Left two nits, hence the hold.

/hold

File: components/ws-manager/pkg/manager/monitor.go (hunk 2 above)
Comment (sagor999) [reply components/ws-manager/pkg/manager/monitor.go:514]: Shouldn't this add it?
https://github.com/gitpod-io/gitpod/blob/f636456dcfd32a13b09b0f3b8e4f177f646a1dc6/components/ws-manager/pkg/manager/monitor.go#L515-L517

**Review by jankeromnes — APPROVED** [review]
Hi! 👋

I have no idea how to test this (also, the "how to test" instructions are empty) but I'm happy to rubber-stamp-approve the 2-lines removed from `components/common-go/kubernetes/kubernetes.go` (co-owned by team WebApp).

File: components/ws-manager/pkg/manager/monitor.go (hunk 2 above)
Comment (csweichel) [reply components/ws-manager/pkg/manager/monitor.go:514]: it does - my bad

**Review by mads-hartmann — COMMENTED** [review]
Awesome - thanks a lot for following up on this 🎉

**Review by jenting — APPROVED** [review]
